src/open-timestamps.js

```diff
@@ -97,6 +97,71 @@ module.exports = {
     return JSON.stringify(json)
   },
 
+  /*
+  * Delete all the dead branches
+  */
+  pruneTree (timestamp) {
+    let prun = timestamp.attestations.length === 0
+    let changed = false
+
+    for (const element of timestamp.ops) {
+      var result = this.pruneTree(element[1])
+      var stampPrunable = result.prunable
+      var stampChanged = result.change
+      changed = changed || stampChanged || stampPrunable
+      if (stampPrunable) {
+        timestamp.ops.delete(element[0])
+      } else {
+        prun = false
+      }
+    }
+    return { prunable: prun, change: changed }
+  },
+
+  /*
+  * Delete all the leaves which didn't match the height passed
+  */
+  discard_attestation (timestamp, heightToKeep) {
+    timestamp.getAttestations().forEach(opStamp => {
```

Comment:
getAttestations() return a Set of every transaction walking the timestamp tree until the leaf. I think in this case, you need to check only the current attestations as: `timestamp.attestations`